lib/kafka/client.rb

```diff
@@ -296,14 +296,16 @@ def producer(
     #
     # @see AsyncProducer
     # @return [AsyncProducer]
-    def async_producer(delivery_interval: 0, delivery_threshold: 0, max_queue_size: 1000, **options)
+    def async_producer(delivery_interval: 0, delivery_threshold: 0, max_queue_size: 1000, max_retries: -1, retry_backoff: 0, **options)
       sync_producer = producer(**options)
 
       AsyncProducer.new(
         sync_producer: sync_producer,
         delivery_interval: delivery_interval,
         delivery_threshold: delivery_threshold,
         max_queue_size: max_queue_size,
+        max_retries: max_retries,
```

Comment:
Should we add these options into the sync producer too?

Reply:
We have `@max_retries` in the `sync_producer` (it will stop after all retries) but the `async_producer` (in the current version) still send `deliver_messages` in an endless loop to the `sync_producer`